src/util/sighash.rs

```diff
@@ -164,7 +165,7 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Error::IoError => write!(f, "IoError"),
+            Error::Io(ref e) => write!(f, "{:?}", e),
```

Comment:
nit: context could be given here, ideally some message that can be grepped when a panic is thrown up-stack. This makes tracing back the root cause of errors much easier.

Reply:
like this? `write!(f, "sighash::Error::Io {:?}", e)`

Reply:
For example, or just a more verbal description like the other variants.

Reply:
using `write!(f, "Writer errored: {:?}", e)` amended last commit to 7b60763075a974ddd4812000ec90be4e17d3fb12